app/models/user.rb

```diff
@@ -1,24 +1,43 @@
 class User < ApplicationRecord
-  has_secure_password
-
+  attr_accessor :remember_token
   before_save :downcase_email
 
   validates :name, presence: true
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}, uniqueness: true
+  has_secure_password
+
+  def authenticated? remember_token
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
+  end
 
   private
```

Comment:
move the private methods to after defined class method
ex:
```suggestion
  class << self
    def new_token
      SecureRandom.urlsafe_base64
    end
    ...
  end
  
  private
  
  def downcase_email
    self.email.downcase!
  end
```